### PR TITLE
Vue: Remove deprecated vue packages from next

### LINKED ISSUE #24779: deprecate @storybook/vue in 7.6
Vue2 reaches EOL on 2023-12-31, and we will remove support for it in Storybook in 8.0. Ahead of that, we should deprecate @storybook/vue in 7.6

#### Tasks
- [x] write up rationale for deprecating @storybook/vue as an RFC in GitHub discussion 
- [ ] deprecate @storybook/vue in 7.6 and point to 👆 the discussion

### PR DESCRIPTION
**Pull request by kasperpeulen**
Fixes #24779 

## What I did

Remove deprecated vue packages from next, as otherwise we keep publishing new version on the next tag.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by JReinhold — CHANGES_REQUESTED** [review]
There's more stuff to remove, eg. the CLI generator for Vue2.

Take a look at https://github.com/storybookjs/storybook/compare/next...remove-deprecated-packages, I also took a stab at this but never got around to opening a PR. (I might also not have gotten everything though, or removed too much, so just look at it for inspiration)

File: code/lib/cli/src/helpers.ts [hunk 1]
```diff
@@ -206,8 +206,7 @@ const frameworkToRenderer: Record<SupportedFrameworks | SupportedRenderers, Supp
   solid: 'solid',
   svelte: 'svelte',
   sveltekit: 'svelte',
-  vue: 'vue',
-  vue3: 'vue',
+  vue3: 'vue3',
```
Comment (JReinhold) [inline code/lib/cli/src/helpers.ts:209]: I'm not sure what this map is used for, but is this change correct?

**Review by JReinhold — APPROVED** [review]
LGTM!

File: code/lib/cli/src/helpers.ts (hunk 1 above)
Comment (ndelangen) [reply code/lib/cli/src/helpers.ts:209]: Looks to me like this  code is only used here:
https://github.com/storybookjs/storybook/blob/48694d15a9de7e9f8a56c7d63f05063490aa8c54/code/lib/cli/src/helpers.ts#L268-L271

It's just modifying the `Configure.mdx` a little bit with some string replacement.

I think it should not mention `vue3`, but rather say `vue`

Comment (kasperpeulen) [reply code/lib/cli/src/helpers.ts]: Thanks!

**Pull request merged**